BHoM_Engine/Compute/RecordEvent.cs

```diff
@@ -46,6 +46,22 @@ public static bool RecordEvent(string message, EventType type = EventType.Unknow
 
         /***************************************************/
 
+        [Description("Record an event with details of a C# exception within the BHoM logging system.")]
+        [Input("exception", "The C# exception being caught to provide the event and stack information for.")]
+        [Input("message", "An optional additional message which will be displayed first in the event log.")]
+        [Output("success", "True if the event has been successfully recorded as a BHoM Event.")]
+        public static bool RecordEvent(Exception exception, string message = "")
+        {
+            if (!string.IsNullOrEmpty(message))
+                message = $"{message}\n\n{exception.ToString()}";
+            else
+                message = exception.ToString();
+
+            return RecordEvent(new Event { Message = message, StackTrace = exception.StackTrace, Type = EventType.Unknown });
```

Comment:
Would get out the message from the exception here, including inner exception, as well as providing settable event type as a parameter, so this can be called by the other methods.

```suggestion
                public static bool RecordEvent(Exception exception, string message = "", EventType type = EventType.Unknown)
        {
            string exceptionMessage = "";

            Exception e = exception;
            while (e != null)
            { 
                exceptionMessage += e.Message + Environment.NewLine;
                e = e.InnerException;
            }

            if (!string.IsNullOrEmpty(message))
                message = $"{message}\n\n{exceptionMessage}";
            else
                message = exceptionMessage;

            return RecordEvent(new Event { Message = message, StackTrace = exception.StackTrace, Type = type });

```

Also need to update the method bellow, that logs the events, as that currently is overwriting the stack trace